ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -113,7 +112,7 @@
   private final Condition pauseCondition = lock.newCondition();
   private final ReplicaSyncUpManager replicaSyncUpManager;
   private final int maxReplicaCountPerRequest;
-  private final AccountService accountService;
+  private final Predicate predicate;
```

Comment:
Could we declare it as Predicate<MessageInfo> and rename it skipPredicate to make it clear we're using it to filter out?

Reply:
Updated.